src/background_client.cpp

```diff
@@ -0,0 +1,578 @@
+/*
+ * Copyright © 2022 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "egfullscreenclient.h"
+#include "background_client.h"
+
+#include "mir/log.h"
+
+#include <chrono>
+#include <cstring>
+#include <codecvt>
+#include <sstream>
+#include <thread>
+
+#include <boost/throw_exception.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/replace.hpp>
+
+#include <miral/configuration_option.h>
```

Comment:
We don't use this here!

```suggestion
#include <mir/fatal.h>
```